source/automated-deployment.rst

```diff
@@ -0,0 +1,131 @@
+        su - {{ BEAMLINE_USER }}
+        source activate collection
+        python -c "from pyOlog.cli.ipy import olog; import os; olog(msg=$1, logbooks='DAMA Software')"
+
+    }
+1. Commit all python files that are in {{ IPYTHON_CONFIGURATION_LOCATION }}/profile_{{ IPYTHON_CONFIGURATION_NAME }} ::
+
+    su - {{ BEAMLINE_USER }}
+    pushd $HOME/{{ IPYTHON_CONFIGURATION_LOCATION }}/profile_{{ IPYTHON_CONFIGURATION_NAME }}
+    find -name *.py -exec git add {} \;
+    git commit -am "Automated weekly code backup: `date +%Y-%W`"
+    git tag -a YYYY-Week -m "Automated weekly tag: `date +%Y-%W`"
+    git push
+    git push --tags
+    # grab a list of the files that were changed
+    FILES=`git show --pretty="format:" --name-only HEAD`
+    if [ -z $FILES ]; then
+        log_to_olog "No untracked or uncommitted files found in the ipython configuration. Yay!" "DAMA Software"
+    else
+        log_to_olog "The following files were found in the ipython configuration: $FILES" "DAMA Software"
+    popd
+    exit
+
+2. Write the current state of the collection environment ::
+
+    conda_envs_path="/opt/conda_envs"
+    previous_collection_name=`ls -r $conda_envs_path | grep collection | head -n2 | tail -n1`
+    conda list -p $conda_envs_path/$previous_collection_name > /opt/conda/envs_list/previous_collection_name--END
+    mkdir -p /opt/conda/envs_list
+    pushd /opt/conda/envs_list
+    if [ -s ${previous_collection_name}--START ]; then
+        diff ${previous_collection_name}--START ${previous_collection_name}--END > ${previous_collection_name}--DIFF
+        if [ -s ${previous_collection_name}--DIFF ]; then
+            # notify the slack channel that the environments are different
+            notify_slack \
+                "Collection+Environment+Changed:+{{ BEAMLINE_USER }}" \
+                `cat /opt/conda/envs_list/${previous_collection_name}--DIFF` \
+                "#automated-deployment"
+        fi
+    else
+        # notify the slack channel that there is no previous environment list to compare
+        notify_slack \
+            "No+Previous+Environment+List:+{{ BEAMLINE_USER }}" \
+            `ls /opt/conda/envs_list/` \
+            "#automated-deployment"
+    fi
+
+3. Create new conda collection environment at BEAMLINE_WORKSTATION with the
+   following command ::
+
+    new_collection_name=collection-`date +%Y-%W`
+    conda create -p /opt/conda_envs/$new_collection_name COLLECTION_{{ BEAMLINE }} --override-channels -c anaconda -c nsls2-{{ BEAMLINE }}
+    # export the list of the new packages
+    conda list -p /opt/conda/envs/$new_collection_name > /opt/conda/envs_list/${new_collection_name}--START
+
+    # log that this conda collection environment was created
+    log_to_olog "New collection environment created at /opt/conda/envs/$new_collection_name" "DAMA Software"
+
+4. Update the collection launching script to point to /opt/conda_envs/collection-YEAR-WEEK_OF_YEAR ::
```

Comment:
:+1: for coming up with a standard launching script. I prefer the self-describing explicitness of the long way, but this is an uphill battle. I hate having to type `iophyd` at XPD and `icollection` somewhere else, etc. Let's just own one way of doing this, however silly it might be.